hrm-domain/hrm-service/migrations/20240329215610-sanitize-identifiers-remove-blank-spaces.js

```diff
@@ -19,17 +19,65 @@
     `);
-    console.log('Sequence "Profiles_id_seq" created');
+    console.log('Contacts fixed');
+
+    await queryInterface.sequelize.query(`
+      DELETE FROM "Profiles" WHERE "id" IN (
+        SELECT "p2i"."profileId" AS "conflictProfileId"
+        FROM (
+          --- All identifiers with conflicts, plus the "target id" which is the minimal id that matches the sanitized identifier
+          SELECT "main"."id" AS "conflictId", "main"."accountSid", "grouped"."targetId", "p2i"."profileId" AS "targetProfile"
+            FROM "Identifiers" "main"
+            INNER JOIN (
+              SELECT "accountSid", replace("identifier", ' ', '') AS "sanitized", MIN("id") AS "targetId" FROM "Identifiers" GROUP BY "accountSid", replace("identifier", ' ', '') HAVING COUNT(*) > 1
+            ) "grouped" ON replace("main"."identifier", ' ', '')="grouped"."sanitized" AND "main"."accountSid"="grouped"."accountSid"
+            INNER JOIN "ProfilesToIdentifiers" "p2i" ON "p2i"."identifierId" = "grouped"."targetId"
+            ORDER BY replace("identifier", ' ', '')
+        ) AS "sanitized"
+        INNER JOIN "ProfilesToIdentifiers" "p2i" ON "p2i"."identifierId" = "sanitized"."conflictId" AND "p2i"."identifierId" != "sanitized"."targetId"
+      )
+    `);
+    console.log('Conflicting profiles deleted');
+
+    await queryInterface.sequelize.query(`
+      DELETE FROM "Identifiers" WHERE "id" IN (
+        SELECT "idx"."id" AS "conflictIdentifierId"
+        FROM (
+          --- All identifiers with conflicts, plus the "target id" which is the minimal id that matches the sanitized identifier
+          SELECT "main"."id" AS "conflictId", "grouped"."targetId"
+            FROM "Identifiers" "main"
+            INNER JOIN (
+              SELECT "accountSid", replace("identifier", ' ', '') AS "sanitized", MIN("id") AS "targetId" FROM "Identifiers" GROUP BY "accountSid", replace("identifier", ' ', '') HAVING COUNT(*) > 1
+            ) "grouped" ON replace("main"."identifier", ' ', '')="grouped"."sanitized" AND "main"."accountSid"="grouped"."accountSid"
+            ORDER BY replace("identifier", ' ', '')
+        ) AS "sanitized"
+        INNER JOIN "Identifiers" "idx" ON "idx"."id" = "sanitized"."conflictId" AND "idx"."id" != "sanitized"."targetId"
+      )
+    `);
+    console.log('Conflicting identifiers deleted');
+
+    await queryInterface.sequelize.query(`
+      UPDATE "Identifiers" "idx" SET "identifier" = replace("idx"."identifier", ' ', '')
+      FROM (
+        SELECT DISTINCT identifiers.* FROM "Identifiers" identifiers
+              JOIN "Contacts" contacts ON identifiers.id = contacts."identifierId"
+              WHERE identifiers.identifier LIKE '% %' AND contacts."channel" IN ('voice', 'whatsapp', 'sms', 'modica')
+      ) AS "whitespaced" WHERE "idx"."id" = "whitespaced"."id"
+    `);
+    console.log('Whitespaced identifiers sanitized');
```

Comment:
@stephenhand Note that here, because `AND contacts."channel" IN ('voice', 'whatsapp', 'sms', 'modica')`, blank-spaced identifiers that already exist in the DBs will not be reachable. I am not sure if this `WHERE` chunk should be present, as I don't think we can, but if we want to keep it we also need to tweek the "search contacts by identifier" endpoint to include the channel type of the task (since we'll want to remove white-spaces only from the above channels).

What do you think? Better to remove this and force non-white-spaced identifiers?